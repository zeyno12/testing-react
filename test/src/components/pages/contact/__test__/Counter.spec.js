import {render,fireEvent,screen} from "@testing-library/react";
import renderer from "react-test-renderer"
import Counter from "../Contact.jsx"

describe("Counter",()=>{
it("should increment counter", ()=>{
    render (<Counter/>)
    const counter= screen.getByTestId("counter")
    const incrementBtn=screen.getByTestId("increment")
    fireEvent.click(incrementBtn)
    expect(counter).toHaveTextContent("2")
})
it("should decrement counter", ()=>{
    render (<Counter/>)
    const counter= screen.getByTestId("counter")
    const decrementBtn=screen.getByTestId("decrement")
    fireEvent.click(decrementBtn)
    expect(counter).toHaveTextContent("0")
})
})

describe('Counter Snapshot', () => {
  it("should matches dom Snapshot",()=>{
const tree=renderer.create(<Counter/>).toJSON();
expect(tree).toMatchSnapshot()
})
})
