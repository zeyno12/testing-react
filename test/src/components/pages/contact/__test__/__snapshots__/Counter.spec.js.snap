// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Snapshot should matches dom Snapshot 1`] = `
<div
  className="counter"
>
  <button
    data-testid="increment"
    onClick={[Function]}
  >
    +
  </button>
  <p
    data-testid="counter"
  >
    1
  </p>
  <button
    data-testid="decrement"
    onClick={[Function]}
  >
    -
  </button>
</div>
`;
